packages/js/src/utils/readApiConnection.ts

```diff
@@ -0,0 +1,268 @@
+import type {
+  GetAssetProofRpcInput,
+  GetAssetProofRpcResponse,
+  GetAssetRpcInput,
+  GetAssetsByOwnerRpcInput,
+  GetAssetsByGroupRpcInput,
+  ReadApiAsset,
+  ReadApiAssetList,
+} from '@/types/ReadApi';
+
+type JsonRpcParams<ReadApiMethodParams> = {
+  method: string;
+  id?: string;
+  params: ReadApiMethodParams;
+};
+
+type JsonRpcOutput<ReadApiJsonOutput> = {
+  result: ReadApiJsonOutput;
+};
+
+export const toNftEditionFromReadApiAsset = (
+  input: ReadApiAsset
+): NftOriginalEdition => {
+  return {
+    model: 'nftEdition',
+    isOriginal: true,
+    address: new PublicKey(input.id),
+    supply: toBigNumber(input.supply.print_current_supply),
+    maxSupply: toBigNumber(input.supply.print_max_supply),
+  };
+};
+
+export const toMintFromReadApiAsset = (input: ReadApiAsset): Mint => {
+  const currency: SplTokenCurrency = {
+    symbol: 'Token',
+    decimals: 0,
+    namespace: 'spl-token',
+  };
+
+  return {
+    model: 'mint',
+    address: new PublicKey(input.id),
+    // TODO(jon): Presumably, this should be the Master Edition address upon decompression
+    mintAuthorityAddress: new PublicKey(input.id),
+    // TODO(jon): Presumably, this should be the Master Edition address upon decompression
+    freezeAuthorityAddress: new PublicKey(input.id),
+    decimals: 0,
+    supply: amount(1, currency),
+    isWrappedSol: false,
+    currency,
+  };
+};
+
+export const toMetadataFromReadApiAsset = (input: ReadApiAsset): Metadata => {
+  const updateAuthority = input.authorities?.find((authority) =>
+    authority.scopes.includes('full')
+  );
+
+  const collection = input.grouping.find(
+    ({ group_key }) => group_key === 'collection'
+  );
+
+  return {
+    model: 'metadata',
+    // TODO(jon): We technically don't have a metadata address anymore. We can derive one though
+    address: Pda.find(BUBBLEGUM_PROGRAM_ID, [
+      Buffer.from('asset', 'utf-8'),
+      new PublicKey(input.compression.tree).toBuffer(),
+      Uint8Array.from(new BN(input.compression.leaf_id).toArray('le', 8)),
+    ]),
+    mintAddress: new PublicKey(input.id),
+    updateAuthorityAddress: new PublicKey(updateAuthority!.address),
+
+    name: input.content.metadata?.name ?? '',
+    symbol: input.content.metadata?.symbol ?? '',
+
+    json: input.content.metadata,
+    jsonLoaded: true,
+    uri: input.content.json_uri,
+    isMutable: input.mutable,
+
+    primarySaleHappened: input.royalty.primary_sale_happened,
+    sellerFeeBasisPoints: input.royalty.basis_points,
+    creators: input.creators,
+
+    editionNonce: input.supply.edition_nonce,
+    tokenStandard: TokenStandard.NonFungible,
+
+    collection: collection
+      ? { address: new PublicKey(collection.group_value), verified: false }
+      : null,
+
+    compression: input.compression,
+
+    // TODO(jon): Read API doesn't return this info
+    collectionDetails: null,
+    // TODO(jon): Read API doesn't return this info
+    uses: null,
```

Comment:
i'm not sure whether or not this is supported, but regardless its not coming through Read API. probably safe to null this out for now